core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -423,18 +432,35 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
     tasks.toMap
   }
 
+  private def getResources(resources: JList[Resource], taskCPUs: Int, taskMemory: Int)
+    : (List[Resource], List[List[Resource]], List[Long]) = {
+    val (afterCPUResources, cpuResourcesToUse) =
+      partitionResources(resources, "cpus", taskCPUs)
+    val (afterMemResources, memResourcesToUse) =
+      partitionResources(afterCPUResources.asJava, "mem", taskMemory)
+    // process port offers
+    val (resourcesWithoutPorts, portResources) = filterPortResources(afterMemResources)
+    val (afterPortResources, portResourcesToUse, portsToUse) =
+      partitionPorts(conf, portResources)
+    (resourcesWithoutPorts ++ afterPortResources,
```

Comment:
Since the tuple its one return value, there is no rule saying i should split it:
https://github.com/databricks/scala-style-guide#doc
I tried to do it but still looks ugly to me:
  private def partitionTaskResources(resources: JList[Resource], taskCPUs: Int, taskMemory: Int)
    : (List[Resource],
       List[List[Resource]],
       List[Long])